src/Security/User/OAuthUserProvider.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * OAuth2 Client Bundle
+ * Copyright (c) KnpUniversity <http://knpuniversity.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace KnpU\OAuth2ClientBundle\Security\User;
+
+use Symfony\Component\Security\Core\Exception\UnsupportedUserException;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+
+class OAuthUserProvider implements UserProviderInterface
+{
+    private $roles;
+
+    public function __construct(array $roles = ['ROLE_USER', 'ROLE_OAUTH_USER'])
```

Comment:
I noticed this wasn't actually configurable in the service. And... I'm not sure if it should be anyways :). After all, this is a helper class to make things easy... and not something the user is bound to. With this setup, they could just create their *own* service for this class. I'm just thinking aloud... I *think* like it how it is now